var/spack/repos/builtin/packages/libxml2/package.py

```diff
@@ -213,6 +217,9 @@ def configure_args(self):
         else:
             args.append("--without-python")
 
+        args.extend(self.enable_or_disable("shared"))
+        args.extend(self.with_or_without("pic"))
```

Comment:
Does the build system handle non-standard compiler pic flags? We may be better off having Spack add `self.compiler.pic_flag` if the package doesn't support pic flags for compilers like fujitsu/ibm/cray (`%fj/%xl/%cce`)

Reply:
It has a pretty extensive list but I don't know that it supports those specific compilers... Let me see if setting CFLAGS will work.

Reply:
```
==> Error: AttributeError: 'AutotoolsBuilder' object has no attribute 'compiler'

The 'libxml2' package cannot find an attribute while trying to build from sources. This might be due to a change in Spack's package format to support multiple build-systems for a single package. You can fix this by updating the build recipe, and you can also report the issue as a bug. More information at https://spack.readthedocs.io/en/latest/packaging_guide.html#installation-procedure

/n/AddMainSpackPackages/spack-main/var/spack/repos/builtin/packages/libxml2/package.py:218, in configure_args:
        215            args.append("--without-python")
        216
        217        if self.spec.satisfies("+pic"):
  >>    218            args.append("CFLAGS=%s" % self.compiler.pic_flag)
        219        else:
        220            args.append("--without-pic")
        221
```

Can you recommend a fix? If there's not an easy solution, since the current default behavior of the package itself is that "--with-pic" is enabled, would it be alright to merge as-is and leave it for later if someone wants to add pic support for other compilers?

Reply:
The best option would probably be in the package class (not the builder):

```
    def flag_handler(self, name, flags):
        if name == "cflags" and self.spec.satisfies("+pic"):
            flags.append(self.compiler.cc_pic_flag)
        return (flags, None, None)
```

Reply:
Added "-DPIC" for consistency with the internal scheme. Tested various variant combinations.